lldb-eval/parser.cc

```diff
@@ -1822,11 +1825,19 @@ ExprResult Parser::ParseIntegerLiteral(clang::NumericLiteralParser& literal,
 
   auto [type, is_unsigned] = PickIntegerType(literal, raw_value);
 
+  bool is_zero_literal =
```

Comment:
How about something like this?

```c++
bool is_zero_literal = raw_value.isNullValue() && value.IsInteger();
```

Also, what are other options for `type`? Can it be anything else apart from the "integer"?

Reply:
Thanks for the suggestion, that looks much cleaner!
I can't think of anything else apart "integer" (`false` and real numbers can't be used in this context), except for `#define`s to literal zero (e.g. `NULL`).

Just "integer" also isn't the best option, e.g. char literal `'\0'` cannot be used in this context, but char literals aren't supported currently, so I added a `TODO`.

Reply:
I mean this is `Parser::ParseIntegerLiteral`, which parses _only_ integer literals. Boolean, floats, chars and so on are handled in other places. So it seems to me here `value.IsInteger()` is guaranteed to be true (e.g. `PickIntegerType` returns only integers types by design).

Reply:
Totally forgot about that. I reduced it to just `raw_value.isNullValue()`.